Automattic/woocommerce-payments
# Add merchant branding settings to the Settings API

Fixes #3018

This PR adds new settings to the **Settings API** to take care of merchant branding on in-person payment receipts. Later, the new settings will be added to the WCPay Settings frontend page, which relies on the Settings API to fetch and save all settings. See the linked issue for details about the fields.

Note that these settings are not merely options in the `wp_options` table. They are primarily fetched from and save in the connected Stripe account. But they may be saved in wp_options for caching.


#### Testing instructions

You may need to set up auth to test the following endpoints. See PbTz5e-12J-p2.

* Send a GET request to the `/payments/settings` endpoint. Verify that the updated settings is returned.

<img width="1502" alt="Screenshot 2021-11-24 at 13 31 27" src="https://user-images.githubusercontent.com/82826872/143230843-e6dc2486-3334-4c62-a053-747ff631c4ab.png">

* Send a PUT or POST request to the `/payments/settings` endpoint with a new value for `account_business_name`. Verify that the new value for account_business_name setting is reflecting in GET response.
Request params ex. :
`
{
   "account_business_name": "John's test Shop",
   "account_business_url": "automattic.com",
    "account_business_support_address": {"city": "test city", "country": "US", "line1": "test line 1", "line2": "test line 2", 
    "postal_code": "postal_code", "state": "test state"},
    "account_business_support_email": "dan.paun@automattic.com",
    "account_business_support_phone": "+16503880901",
    "account_branding_logo": "file_1JwoLdJ5cIRIG92xJPqLtwQ0",
    "account_branding_icon": "file_1JwoQvJ5cIRIG92xPmf9kMGx",
    "account_branding_primary_color": "#ffffff",
    "account_branding_secondary_color": "#ffffff"
}
`

<img width="1536" alt="Screenshot 2021-11-24 at 13 35 12" src="https://user-images.githubusercontent.com/82826872/143231165-216c6a90-0b00-453e-8917-4a3bf9629fac.png">

Note: `account_branding_logo` and `account_branding_icon` fields support the id of already uploaded files.

----------


**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions)
